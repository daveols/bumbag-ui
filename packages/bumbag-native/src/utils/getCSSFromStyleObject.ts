import _kebabCase from 'lodash/kebabCase';

import {
  altitude,
  border,
  borderRadius,
  font,
  fontSize,
  lineHeight,
  palette,
  space,
  fontWeight,
  letterSpacing,
} from './theme';
import { Platform } from 'react-native';

const FLEX_HORIZONTAL_ALIGN_MAP = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
};

const FLEX_VERTICAL_ALIGN_MAP = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end',
};

const alignAttributes = ['alignX', 'alignY'];
const altitudeAttributes = ['altitude'];
const borderAttributes = ['border'];
const borderRadiusAttributes = [
  'borderRadius',
  'borderLeftRadius',
  'borderRightRadius',
  'borderTopRadius',
  'borderBottomRadius',
  'borderTopLeftRadius',
  'borderBottomLeftRadius',
  'borderTopRightRadius',
  'borderBottomRightRadius',
];
const colorAttributes = [
  'color',
  'backgroundColor',
  'borderBlockEndColor',
  'borderBlockStartColor',
  'borderBottomColor',
  'borderColor',
  'borderInlineEndColor',
  'borderInlineStartColor',
  'borderLeftColor',
  'borderRightColor',
  'borderTopColor',
  'borderBottomColor',
  'caretColor',
  'columnRuleColor',
  'outlineColor',
  'textDecorationColor',
  'textEmphasisColor',
];
const spaceAttributes = [
  'margin',
  'marginLeft',
  'marginRight',
  'marginTop',
  'marginBottom',
  'marginX',
  'marginY',
  'padding',
  'paddingLeft',
  'paddingRight',
  'paddingTop',
  'paddingBottom',
  'paddingX',
  'paddingY',
  'top',
  'left',
  'bottom',
  'right',
  'grid-gap',
  'grid-column-gap',
  'grid-row-gap',
  'height',
  'width',
  'minWidth',
  'minHeight',
  'maxWidth',
  'maxHeight',
];
const fontAttributes = ['font', 'fontFamily'];
const fontSizeAttributes = ['fontSize'];
const fontWeightAttributes = ['fontWeight'];
const lineHeightAttributes = ['lineHeight'];
const letterSpacingAttributes = ['letterSpacing'];

const attributeMaps = {
  font: ['fontFamily'],
  marginY: ['marginTop', 'marginBottom'],
  paddingY: ['paddingTop', 'paddingBottom'],
  marginX: ['marginLeft', 'marginRight'],
  paddingX: ['paddingLeft', 'paddingRight'],
  borderTopRadius: ['borderTopLeftRadius', 'borderTopRightRadius'],
  borderBottomRadius: ['borderBottomLeftRadius', 'borderBottomRightRadius'],
  borderLeftRadius: ['borderTopLeftRadius', 'borderBottomLeftRadius'],
  borderRightRadius: ['borderTopRightRadius', 'borderBottomRightRadius'],
};

const platforms = ['web', 'native', 'ios', 'android'];
function isPlatform(selector) {
  return platforms.includes(selector);
}

const states = ['hover', 'hoveractive', 'focus', 'active'];
function isPseudo(selector) {
  return states.includes(selector);
}

export function getAlignmentStyles({ attribute, value, props }) {
  const alignYAttribute = props.display && props.display.includes('flex') ? 'align-items' : 'justify-content';
  const alignXAttribute = props.display && props.display.includes('flex') ? 'justify-content' : 'align-items';
  return `
      display: flex;

      ${
        !props.display &&
        `
          flex-direction: column;
        `
      }

      ${
        attribute === 'alignY' &&
        `
          ${alignYAttribute}: ${FLEX_VERTICAL_ALIGN_MAP[value]};
        `
      }

      ${
        attribute === 'alignX' &&
        `
          ${alignXAttribute}: ${FLEX_HORIZONTAL_ALIGN_MAP[value]};
        `
      }
    `;
}

function getAltitudeStyles({ theme, value }) {
  const altitudeStyle = altitude(value)({ theme });
  if (altitudeStyle) {
    return altitudeStyle;
  }
  return value;
}

function getBorderValue({ theme, value }) {
  const borderValue = border(value)({ theme });
  if (borderValue) {
    const borderColor = palette(borderValue.color)({ theme });
    return `${borderValue.width} ${borderValue.style} ${borderColor}`;
  }
  return value;
}

function getBorderRadiusValue({ theme, value }) {
  const borderRadiusValue = borderRadius(value)({ theme });
  if (borderRadiusValue) {
    return borderRadiusValue;
  }
  return value;
}

function getColorValue({ colorMode, theme, value }) {
  const color = palette(value)({ colorMode, theme });
  if (color) {
    return color;
  }
  return value;
}

function getSpaceValue({ theme, value }) {
  const spacing = space(value)({ theme });
  if (spacing) {
    return `${spacing}px`;
  }
  return value;
}

function getFontValue({ theme, value }) {
  const fontValue = font(value)({ theme });
  if (fontValue) {
    return fontValue;
  }
  return value;
}

function getFontSizeValue({ theme, value }) {
  const size = fontSize(value)({ theme });
  if (size) {
    return `${size}px`;
  }
  return value;
}

function getFontWeightValue({ theme, value, fontFamily }) {
  const weight = fontWeight({ fontFamily, fontWeight: value })({ theme });
  if (weight) {
    return weight;
  }
  return value;
}

function getLineHeightValue({ theme, value }) {
  const height = lineHeight(value)({ theme });
  if (height) {
    return height;
  }
  return value;
}

function getLetterSpacingValue({ theme, value }) {
  const spacing = letterSpacing(value)({ theme });
  if (spacing) {
    return spacing;
  }
  return value;
}

export function getCSSFromStyleObject(
  props,
  theme,
  colorMode,
  { fromProps = false, disableCSSProps = [], events = {} } = {}
) {
  let style = { ...props };
  if (style) {
    let styleEntries = Object.entries(style);
    styleEntries = styleEntries.reduce((prevStyle, [attribute, value]) => {
      let entries = [[attribute, value]];
      if (attributeMaps[attribute]) {
        entries = attributeMaps[attribute].map((attribute) => [attribute, value]);
      }
      return [...prevStyle, ...entries];
    }, []);

    style = styleEntries.reduce((prevStyle, [attribute, value]) => {
      let newValue = value;
      if (typeof value === 'function') {
        newValue = value({ theme });
      }
      if (attribute.includes(':')) {
        return `
          ${prevStyle ? `${prevStyle};` : ''}

          ${attribute} {
            ${getCSSFromStyleObject(value, theme, colorMode, { fromProps })};
          }
        `;
      }
      if (typeof newValue === 'string') {
        newValue = { default: newValue };
      }
      if (attribute.includes('_')) {
        const selector = attribute.slice(1);
        if (isPlatform(selector)) {
          const platform = selector;
          return `
            ${prevStyle ? `${prevStyle};` : ''}
            ${
              (Platform.OS === platform || (platform === 'native' && Platform.OS !== 'web')) &&
              getCSSFromStyleObject(value, theme, colorMode, { fromProps })
            }
          `;
        }
        if (isPseudo(selector)) {
          return `
            ${prevStyle ? `${prevStyle};` : ''}
            ${events[selector] ? getCSSFromStyleObject(value, theme, colorMode, { fromProps }) : ''}
          `;
        }
      }
      // if (newValue._animation) {
      //   return { [attribute]: newValue };
      // }
      const newStyle = Object.entries(newValue || {}).reduce((prevStyle, [platform, value]) => {
        let newStyle;
        let newValue = value;
        if (alignAttributes.includes(attribute)) {
          newStyle = getAlignmentStyles({ attribute, value, props });
          newValue = undefined;
        }
        if (altitudeAttributes.includes(attribute)) {
          newStyle = getAltitudeStyles({ theme, value });
          newValue = undefined;
        }
        if (borderAttributes.includes(attribute)) {
          newValue = getBorderValue({ theme, value });
        }
        if (borderRadiusAttributes.includes(attribute)) {
          newValue = getBorderRadiusValue({ theme, value });
        }
        if (colorAttributes.includes(attribute)) {
          newValue = getColorValue({ colorMode, theme, value });
        }
        if (spaceAttributes.includes(attribute)) {
          newValue = getSpaceValue({ theme, value });
        }
        if (fontAttributes.includes(attribute)) {
          newValue = '';
        }
        if (fontSizeAttributes.includes(attribute)) {
          if (disableCSSProps.includes('fontSize')) {
            return {};
          }
          newValue = getFontSizeValue({ theme, value });
        }
        if (fontWeightAttributes.includes(attribute)) {
          newValue = getFontWeightValue({ theme, value, fontFamily: props.font });
          const fontFamily = font(props.font || 'default')({ theme });
          if (typeof newValue === 'object') {
            newStyle = `
              font-family: ${fontFamily}${newValue.fontFamilySuffix ? `-${newValue.fontFamilySuffix}` : ''};
            `;
            newValue = newValue.fontWeight;
          } else if (fontFamily) {
            newStyle = `
              font-family: ${fontFamily};
            `;
          }
        }
        if (lineHeightAttributes.includes(attribute)) {
          newValue = getLineHeightValue({ theme, value });
        }
        if (letterSpacingAttributes.includes(attribute)) {
          newValue = getLetterSpacingValue({ theme, value });
        }
        if (platform === 'default' || Platform.OS === platform || (platform === 'native' && Platform.OS !== 'web')) {
          // @ts-ignore
          return `
            ${newStyle ? `${newStyle};` : ''}
            ${prevStyle ? `${prevStyle};` : ''}
            ${
              newValue
                ? `
                  ${_kebabCase(attribute)}: ${newValue};
                `
                : ''
            }
          `;
        }
        return `
          ${newStyle ? `${newStyle};` : ''}
          ${prevStyle ? `${prevStyle};` : ''}
        `;
      }, ``);
      return `
        ${newStyle ? `${newStyle};` : ''}
        ${prevStyle ? `${prevStyle};` : ''}
      `;
    }, ``);
  }

  return style;
}
