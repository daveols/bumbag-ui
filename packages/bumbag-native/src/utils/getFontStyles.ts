import { font, fontWeight as getFontWeight } from './theme';

export function getFontStyles({ fontWeight: _fontWeight, fontFamily: _fontFamilyThemeKey = 'default' }) {
  return (props) => {
    const fontFamilyThemeKey = props.font || _fontFamilyThemeKey;
    const fontWeight = getFontWeight({ fontFamily: fontFamilyThemeKey, fontWeight: _fontWeight })(props);
    const fontFamily = font(fontFamilyThemeKey)(props);

    if (typeof fontWeight === 'object') {
      const newFontFamily = `${fontFamily}${fontWeight.fontFamilySuffix ? `-${fontWeight.fontFamilySuffix}` : ''}`;
      const newFontWeight = fontWeight.fontWeight || 'normal';
      return `
        font-family: ${newFontFamily};
        font-weight: ${newFontWeight};
      `;
    }

    return `
      ${fontFamily ? `font-family: ${fontFamily};` : ''}
      font-weight: ${fontWeight};
    `;
  };
}
