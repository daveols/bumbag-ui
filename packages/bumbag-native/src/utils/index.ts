export * from './breakpoint';
export * from './getCSSFromStyleObject';
export * from './getFontStyles';
export * from './theme';
export * from './throttle';
export * from './useBreakpoint';
export * from './useStyle';
export * from './useTheme';
