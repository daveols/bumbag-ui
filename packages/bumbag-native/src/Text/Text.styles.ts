import { Animated, Platform, Text } from 'react-native';
import { styled } from '../styled';
import { palette, theme } from '../utils/theme';

function getTextStyles(props) {
  return `
    color: ${palette('text')(props)};

    ${
      Platform.OS === 'web'
        ? `
            font-size: unset;
          `
        : ''
    }

    ${theme('native.Text', 'styles.base')(props)};
  `;
}

export const StyledText = styled(Text)`
  ${getTextStyles}
`;

export const StyledAnimatedText = styled(Animated.Text)`
  ${getTextStyles}
`;
