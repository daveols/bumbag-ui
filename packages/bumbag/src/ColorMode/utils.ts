import { omit } from '../utils/omit';

const rootElementClassPrefix = 'bb-mode';
const cssVariablePrefix = '--bb';
const palettePrefix = `${cssVariablePrefix}-palette`;

export function mapCSSVariables(obj) {
  return Object.entries(obj).reduce((cssVariables, [key, value]) => {
    return {
      ...cssVariables,
      [`${palettePrefix}-${key}`]: value,
    };
  }, {});
}

export function getColorModesCSSVariables(theme) {
  if (!theme.palette) {
    return {
      [theme.cssVariablesSelector]: {},
    };
  }
  let cssVariables = mapCSSVariables(omit(theme.palette, 'modes'));
  cssVariables = Object.entries(theme.palette.modes || {}).reduce((cssVariables, [modeKey, value]) => {
    return {
      ...cssVariables,
      [`&.${rootElementClassPrefix}-${modeKey}`]: mapCSSVariables(value),
    };
  }, cssVariables);
  return {
    [theme.cssVariablesSelector]: {
      ...cssVariables,
    },
  };
}

export function getColorFromCSSVariable(selector, fallback) {
  return `var(${palettePrefix}-${selector}, ${fallback})`;
}

export function addColorModeRootElementClassName(nextMode: string, prevMode?: string) {
  const rootElement = window?.document?.getElementsByTagName?.('html')?.[0] ?? window?.document?.body;
  if (rootElement) {
    if (prevMode) {
      rootElement.classList.remove(`${rootElementClassPrefix}-${prevMode}`);
    }
    rootElement.classList.add(`${rootElementClassPrefix}-${nextMode}`);
  }
}

export function getDefaultColorMode(mode, { localStorage, theme }) {
  const { useSystemColorMode } = theme.modes || {};
  let defaultMode = mode;
  if (typeof window !== 'undefined') {
    if (useSystemColorMode && window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
      defaultMode = 'dark';
    }
    if (localStorage.get('mode')) {
      defaultMode = localStorage.get('mode');
    }
  }
  return defaultMode;
}
