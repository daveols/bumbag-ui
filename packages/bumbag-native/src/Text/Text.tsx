import { TextProps as RNTextProps } from 'react-native';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import * as styles from './Text.styles';

export type LocalTextProps = {};
export type TextProps = BoxProps & RNTextProps & LocalTextProps;

const useProps = createHook<TextProps>(
  (props) => {
    const boxProps = Box.useProps(props);
    return { ...boxProps };
  },
  { defaultProps: { fontSize: '200' }, themeKey: 'native.Text' }
);

export const Text = createComponent<TextProps>(
  (props) => {
    const textProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledText,
      htmlProps: { ...textProps, font: props.font || props.fontFamily },
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.Text',
    },
    themeKey: 'native.Text',
  }
);
