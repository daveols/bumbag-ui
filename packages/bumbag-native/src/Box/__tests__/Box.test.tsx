import * as React from 'react';
import { applyTheme } from 'bumbag/utils/applyTheme';
import { Box } from '../index';
import { Text } from '../../Text';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Box />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should assign a ref', () => {
    const ref = React.createRef();
    render(
      <Box ref={ref}>
        <Text>Hello world</Text>
      </Box>
    );
    expect(ref.current).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Box backgroundColor="red" color="white" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with height/width CSS props', () => {
    const { container } = render(<Box height="16px" width="32px" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with height/width CSS props', () => {
    const { container } = render(<Box height="major-2" width="minor-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for an altitude', () => {
    const { container } = render(<Box altitude="400" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a color CSS prop', () => {
    const { container } = render(<Box backgroundColor="white900" color="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a spacing CSS prop', () => {
    const { container } = render(<Box margin="major-4" paddingLeft="major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with negative spacing CSS props', () => {
    const { container } = render(<Box margin="-major-4" paddingLeft="-major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a spacing CSS prop (marginX, paddingX)', () => {
    const { container } = render(<Box marginX="major-4" paddingX="major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a spacing CSS prop (marginY, paddingY)', () => {
    const { container } = render(<Box marginY="major-4" paddingY="major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a font size CSS prop', () => {
    const { container } = render(<Box fontSize="400" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a font weight CSS prop', () => {
    const { container } = render(<Box fontWeight="600" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for a custom border', () => {
    const { container } = render(<Box border="default" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  ['default', '0', '1', '2', '3', '4', '5', '6'].forEach((borderRadius) => {
    it(`should render correctly for a borderRadius of ${borderRadius}`, () => {
      const { container } = render(<Box borderRadius={borderRadius} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render correctly with a color CSS prop', () => {
    const { container } = render(<Box backgroundColor="white900" color="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a spacing CSS prop', () => {
    const { container } = render(<Box margin="major-4" paddingLeft="major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  ['left', 'right', 'center'].forEach((alignX) => {
    it(`should render correctly for an alignX of ${alignX}`, () => {
      // @ts-ignore
      const { container } = render(<Box alignX={alignX} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  ['top', 'bottom', 'center'].forEach((alignY) => {
    it(`should render correctly for an alignY of ${alignY}`, () => {
      // @ts-ignore
      const { container } = render(<Box alignY={alignY} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Box.base should render correctly', () => {
    const { container } = render(
      <Box>
        <Text>hello world</Text>
      </Box>,
      {
        theme: { Box: { styles: { base: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Box.base should render correctly', () => {
    const { container } = render(
      <Box color="primary">
        <Text>hello world</Text>
      </Box>,
      {
        theme: {
          Box: { styles: { base: (props) => ({ backgroundColor: props.color }) } },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('Box.base should render correctly', () => {
    const { container } = render(
      <Box
        overrides={{
          Box: {
            styles: { base: { backgroundColor: 'red' } },
            defaultProps: {
              altitude: '100',
              padding: 'major-2',
            },
          },
        }}
      >
        <Text>hello world</Text>
      </Box>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('applyTheme', () => {
  it('renders correctly', () => {
    const Card = applyTheme(Box, {
      styles: {
        base: {
          backgroundColor: 'red',
        },
      },
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <Box variant="test">
        <Text>hello world</Text>
      </Box>,
      {
        theme: {
          Box: {
            variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
          },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Box
        overrides={{
          Box: {
            variants: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        variant="test"
      >
        <Text>hello world</Text>
      </Box>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Box, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      variants: {
        test: {
          defaultProps: {
            color: 'red',
          },
        },
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('modes', () => {
  it('should render correctly when colorMode is set globally', () => {
    const { container } = render(
      <Box>
        <Text>hello world</Text>
      </Box>,
      {
        colorMode: 'test',
        theme: {
          Box: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: { color: 'primaryTint' },
              },
            },
          },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode set as prop', () => {
    const { container } = render(
      <Box colorMode="test">
        <Text>hello world</Text>
      </Box>,
      {
        theme: {
          Box: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: { color: 'primaryTint' },
              },
            },
          },
        },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Box
        overrides={{
          Box: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        colorMode="test"
      >
        <Text>hello world</Text>
      </Box>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Box, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      modes: {
        test: {
          styles: { base: { backgroundColor: 'red' } },
          defaultProps: { color: 'primaryTint' },
        },
      },
    });
    const { container } = render(<Card colorMode="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
