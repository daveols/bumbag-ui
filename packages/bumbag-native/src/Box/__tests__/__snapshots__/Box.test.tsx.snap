// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyTheme renders correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px;"
/>
`;

exports[`modes renders correctly for applyTheme 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(238, 237, 253); padding: 16px 16px 16px 16px;"
/>
`;

exports[`modes renders correctly for overrides 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`modes should render correctly when colorMode is set globally 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); color: rgb(238, 237, 253);"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(238, 237, 253); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`modes should render correctly when colorMode set as prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); color: rgb(238, 237, 253);"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`overrides Box.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`props should assign a ref 1`] = `
<div
  class="css-view-1dbjc4n"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
/>
`;

exports[`props should render correctly for a borderRadius of 0 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 0px; border-bottom-right-radius: 0px; border-top-left-radius: 0px; border-top-right-radius: 0px;"
/>
`;

exports[`props should render correctly for a borderRadius of 1 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 2px; border-bottom-right-radius: 2px; border-top-left-radius: 2px; border-top-right-radius: 2px;"
/>
`;

exports[`props should render correctly for a borderRadius of 2 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 4px; border-bottom-right-radius: 4px; border-top-left-radius: 4px; border-top-right-radius: 4px;"
/>
`;

exports[`props should render correctly for a borderRadius of 3 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-left-radius: 6px; border-top-right-radius: 6px;"
/>
`;

exports[`props should render correctly for a borderRadius of 4 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 8px; border-bottom-right-radius: 8px; border-top-left-radius: 8px; border-top-right-radius: 8px;"
/>
`;

exports[`props should render correctly for a borderRadius of 5 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 16px; border-bottom-right-radius: 16px; border-top-left-radius: 16px; border-top-right-radius: 16px;"
/>
`;

exports[`props should render correctly for a borderRadius of 6 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 32px; border-bottom-right-radius: 32px; border-top-left-radius: 32px; border-top-right-radius: 32px;"
/>
`;

exports[`props should render correctly for a borderRadius of default 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; border-top-left-radius: 6px; border-top-right-radius: 6px;"
/>
`;

exports[`props should render correctly for a custom border 1`] = `
<div
  class="css-view-1dbjc4n"
  style="border-top-color: rgba(230,230,235,1.00); border-right-color: rgba(230,230,235,1.00); border-bottom-color: rgba(230,230,235,1.00); border-left-color: rgba(230,230,235,1.00); border-top-style: solid; border-right-style: solid; border-bottom-style: solid; border-left-style: solid; border-top-width: 1px; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px;"
/>
`;

exports[`props should render correctly for an alignX of center 1`] = `
<div
  class="css-view-1dbjc4n"
  style="align-items: center; display: flex; flex-direction: column;"
/>
`;

exports[`props should render correctly for an alignX of left 1`] = `
<div
  class="css-view-1dbjc4n"
  style="align-items: flex-start; display: flex; flex-direction: column;"
/>
`;

exports[`props should render correctly for an alignX of right 1`] = `
<div
  class="css-view-1dbjc4n"
  style="align-items: flex-end; display: flex; flex-direction: column;"
/>
`;

exports[`props should render correctly for an alignY of bottom 1`] = `
<div
  class="css-view-1dbjc4n"
  style="display: flex; flex-direction: column; justify-content: flex-end;"
/>
`;

exports[`props should render correctly for an alignY of center 1`] = `
<div
  class="css-view-1dbjc4n"
  style="display: flex; flex-direction: column; justify-content: center;"
/>
`;

exports[`props should render correctly for an alignY of top 1`] = `
<div
  class="css-view-1dbjc4n"
  style="display: flex; flex-direction: column; justify-content: flex-start;"
/>
`;

exports[`props should render correctly for an altitude 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 6px 24px rgba(0,0,0,0.15);"
/>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); color: rgb(255, 255, 255);"
/>
`;

exports[`props should render correctly with a color CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(217, 217, 225); color: rgb(87, 79, 235);"
/>
`;

exports[`props should render correctly with a color CSS prop 2`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(217, 217, 225); color: rgb(87, 79, 235);"
/>
`;

exports[`props should render correctly with a font size CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="font-size: 24px;"
/>
`;

exports[`props should render correctly with a font weight CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="font-weight: 600;"
/>
`;

exports[`props should render correctly with a spacing CSS prop (marginX, paddingX) 1`] = `
<div
  class="css-view-1dbjc4n"
  style="margin-left: 32px; margin-right: 32px; padding-left: 16px; padding-right: 16px;"
/>
`;

exports[`props should render correctly with a spacing CSS prop (marginY, paddingY) 1`] = `
<div
  class="css-view-1dbjc4n"
  style="margin-bottom: 32px; margin-top: 32px; padding-bottom: 16px; padding-top: 16px;"
/>
`;

exports[`props should render correctly with a spacing CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="margin: 32px 32px 32px 32px; padding-left: 16px;"
/>
`;

exports[`props should render correctly with a spacing CSS prop 2`] = `
<div
  class="css-view-1dbjc4n"
  style="margin: 32px 32px 32px 32px; padding-left: 16px;"
/>
`;

exports[`props should render correctly with height/width CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="height: 16px; width: 32px;"
/>
`;

exports[`props should render correctly with height/width CSS props 2`] = `
<div
  class="css-view-1dbjc4n"
  style="height: 16px; width: 8px;"
/>
`;

exports[`props should render correctly with negative spacing CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="margin: -32px -32px -32px -32px; padding-left: -16px;"
/>
`;

exports[`theming Box.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68);"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`theming Box.base should render correctly 2`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(87, 79, 235);"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`variants renders correctly for applyTheme 1`] = `
<div
  class="css-view-1dbjc4n"
  style="box-shadow: 0px 1px 4px rgba(0,0,0,0.15); color: rgb(239, 68, 68); padding: 16px 16px 16px 16px;"
/>
`;

exports[`variants renders correctly for overrides 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); padding: 16px 16px 16px 16px;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(239, 68, 68);"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(22, 30, 46); font-size: 16px; font-weight: 400;"
  >
    hello world
  </div>
</div>
`;
